### PR TITLE
Add support for Geistmaschine Macropod

### PR DESCRIPTION
**Pull request by ebastler**
This PR adds the Geistmaschine Macropod macropad to QMK.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/geistmaschine/macropod/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] = { ENCODER_CCW_CW(KC_MRWD, KC_MFFD)  },
+    [2] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  },
+    [3] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  },
```
Comment (elpekenin) [inline keyboards/geistmaschine/macropod/keymaps/default/keymap.c:26]:
```suggestion
```

File: keyboards/geistmaschine/macropod/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [2] = LAYOUT_fourkey (
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [3] = LAYOUT_fourkey (
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (elpekenin) [inline keyboards/geistmaschine/macropod/keymaps/default/keymap.c:18]:
```suggestion
```

File: keyboards/geistmaschine/macropod/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] = { ENCODER_CCW_CW(KC_MRWD, KC_MFFD)  },
+    [2] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  },
+    [3] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  },
```
Comment (elpekenin) [inline keyboards/geistmaschine/macropod/keymaps/via/keymap.c:26]:
```suggestion
```

File: keyboards/geistmaschine/macropod/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [2] = LAYOUT_fourkey (
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [3] = LAYOUT_fourkey (
+        KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (elpekenin) [inline keyboards/geistmaschine/macropod/keymaps/via/keymap.c:18]:
```suggestion
```

File: keyboards/geistmaschine/macropod/readme.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Ada
+
+![geist](imgur.com image replace me!)
```
Comment (elpekenin) [inline keyboards/geistmaschine/macropod/readme.md:3]: Missing the image

**Review by elpekenin — CHANGES_REQUESTED** [review]
No need for empty layers

**Comment by ebastler** [comment]
Hmm, picture is too big and has weird color transition. Will replace again.

File: keyboards/geistmaschine/macropod/readme.md (hunk 5 above)
Comment (ebastler) [reply keyboards/geistmaschine/macropod/readme.md]: Thanks, forgot that... Added pic and updated readme in general.

File: keyboards/geistmaschine/macropod/rules.mk [hunk 6]
```diff
@@ -0,0 +1,7 @@
+ENCODER_MAP_ENABLE = yes
+
```
Comment (waffle87) [inline keyboards/geistmaschine/macropod/rules.mk:2]: Same comment from other PR about encoder map not being enabled at keyboard level.
```suggestion
```

Comment (ebastler) [reply keyboards/geistmaschine/macropod/rules.mk]: Sorry for applying the changes so late, have been busy in the past few days - I moved the encoder map enable to keymap level configs, and added an encoder map to board.c!

File: keyboards/geistmaschine/macropod/readme.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# Geistmaschine Macropod
+
+![macropod](https://i.imgur.com/lAsCslA.png)
```
Comment (drashna) [inline keyboards/geistmaschine/macropod/readme.md:3]:
```suggestion
![macropod](https://i.imgur.com/lAsCslAh.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/geistmaschine/macropod/readme.md (hunk 7 above)
Comment (ebastler) [reply keyboards/geistmaschine/macropod/readme.md:3]: Same issue as with the other PR, the picture is already scaled at 640px width, but the imgur compression removes transparency and adds some artifacts around the edges.

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**